app/pages/lab-pages-editor/DataManager.jsx

```diff
@@ -0,0 +1,82 @@
+/*
+Data Manager
+Responsible for fetching, updating, and saving Workflow resources from/to
+the Panoptes service.
+ */
+
+import apiClient from 'panoptes-client/lib/api-client'
+import { useEffect, useState } from 'react'
+import { WorkflowContext } from './context.jsx'
+
+export default function DataManager ({
+  // key: to ensure DataManager renders FRESH (with states reset) whenever workflowId changes, use <DataManager key={workflowId} ... />
+  children,
+  workflowId,
+}) {
+  let initialised = false
+  const [ apiData, setApiData ] = useState({
+    workflow: null,
+    status: 'ready',
+  })
+
+  // Fetch workflow when the component loads for the first time.
+  // See notes about 'key' prop, to ensure states are reset:
+  // https://react.dev/learn/you-might-not-need-an-effect#resetting-all-state-when-a-prop-changes
+  // Also see general pattern notes:
+  // https://react.dev/reference/react/useEffect#fetching-data-with-effects
+  useEffect(async function whenComponentMounts_doInitialise () {
+    console.log('+++ DataManager: fetchWorkflow', workflowId)
+    if (initialised) return 
+    if (!workflowId) return
+
+    try {
+      setApiData({
+        workflow: null,
+        status: 'fetching'
+      })
+
+      const wf = await apiClient.type('workflows').get(workflowId)
```

Comment:
If this is going to be used in the monorepo, it needs to use `@zooniverse/panoptes-js` as its API client.

Reply:
Noted, but for now this will use the classic PJC apiClient to keep in line with PFE. I'm still trying to isolate the the fetch functionality so the PJC can be replaced with zooniverse/panoptes-js if/when we eventually move this code to FEM